libethereum/BlockChain.cpp

```diff
@@ -384,16 +398,19 @@ void BlockChain::rebuild(fs::path const& _path, std::function<void(unsigned, uns
 
     // Manually insert the genesis block details so that they're available during import of the
     // first block.
-    auto const genesisDetails = BlockDetails{0, s.info().difficulty(), h256(), {}};
+    auto const genesisDetails =
+        BlockDetails{0 /* block number */, s.info().difficulty(), h256{} /* parent */,
+            {} /* children */, static_cast<unsigned>(m_params.genesisBlock().size())};
     m_details[m_genesisHash] = genesisDetails;
-    auto const genesisDetailsRlp = genesisDetails.rlp();
     m_extrasDB->insert(
-        toSlice(m_genesisHash, ExtraDetails), (db::Slice)dev::ref(genesisDetailsRlp));
+        toSlice(m_genesisHash, ExtraDetails), (db::Slice)dev::ref(genesisDetails.rlp()));
```

Comment:
I think this is dangerous and might crash. `genesisDetails.rlp()` creates temporary `bytes`, then `dev::ref` creates `vector_ref` that stores the pointer to the data of that temporary, but temporary is not guaranteed to exist after `vector_ref(bytes const&)` is finished